csds-client/client/client_test.go

```diff
@@ -0,0 +1,171 @@
+package client
+
+import (
+	envoy_service_status_v2 "github.com/envoyproxy/go-control-plane/envoy/service/status/v2"
+
+	"bytes"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/encoding/prototext"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+)
+
+// test parsing -request_file to nodematcher
+func TestParseNodeMatcherWithFile(t *testing.T) {
+	c := Client{
+		info: Flag{
+			requestFile: "./test_request.yaml",
+		},
+	}
+	err := c.parseNodeMatcher()
+	if err != nil {
+		t.Errorf("Parse NodeMatcher Error: %v", err)
+	}
+	if c.nm == nil {
+		t.Errorf("Parse NodeMatcher Failure!")
+	}
+	want := "node_id:{exact:\"fake_node_id\"} node_metadatas:{path:{key:\"TRAFFICDIRECTOR_GCP_PROJECT_NUMBER\"} value:{string_match:{exact:\"fake_project_number\"}}} node_metadatas:{path:{key:\"TRAFFICDIRECTOR_NETWORK_NAME\"} value:{string_match:{exact:\"fake_network_name\"}}}"
+	get, err := prototext.Marshal(c.nm[0])
+	if err != nil {
+		t.Errorf("Parse NodeMatcher Error: %v", err)
+	}
+	getStr := string(get)
```

Comment:
You probably don't need this extra var, just cast to string inline if needed.

Reply:
Solved.